File: BetterLyrics.Core/Models/Settings/AppSettings.cs
using BetterLyrics.Core.Collections;
using BetterLyrics.Core.Helpers;
using BetterLyrics.Core.Models.Entities;
using CommunityToolkit.Mvvm.ComponentModel;

namespace BetterLyrics.Core.Models.Settings
{
    public partial class AppSettings : ObservableRecipient
    {
        public string Version { get; set; } = MetadataHelper.AppVersion;

        [ObservableProperty][NotifyPropertyChangedRecipients] public partial TranslationSettings TranslationSettings { get; set; } = new TranslationSettings();
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial GeneralSettings GeneralSettings { get; set; } = new GeneralSettings();
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial MusicGallerySettings MusicGallerySettings { get; set; } = new MusicGallerySettings();
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial Core.Models.Settings.AdvancedSettings AdvancedSettings { get; set; } = new Core.Models.Settings.AdvancedSettings();
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial LyricsSaveConfig LyricsSaveConfig { get; set; } = new LyricsSaveConfig();
        [ObservableProperty] public partial SystemTraySettings SystemTraySettings { get; set; } = new SystemTraySettings();

        [ObservableProperty][NotifyPropertyChangedRecipients] public partial FullyObservableCollection<Core.Models.Settings.MediaFolder> LocalMediaFolders { get; set; } = [];
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial FullyObservableCollection<MediaSourceProviderInfo> MediaSourceProvidersInfo { get; set; } = [];
        [Obsolete][ObservableProperty][NotifyPropertyChangedRecipients] public partial FullyObservableCollection<MappedSongSearchQuery> MappedSongSearchQueries { get; set; } = [];
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial FullyObservableCollection<LyricsWindowStatus> WindowBoundsRecords { get; set; } = [];
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial FullyObservableCollection<SongsTabInfo> StarredPlaylists { get; set; } = [];
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial FullyObservableCollection<PluginInfo> PluginsInfo { get; set; } = [];
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial FullyObservableCollection<LyricsCardConfig> LyricsCardConfigs { get; set; } = [];
        [ObservableProperty][NotifyPropertyChangedRecipients] public partial FullyObservableCollection<LayoutProfile> LayoutProfiles { get; set; } = [];

        public AppSettings() { }
    }
}

File: src/BetterLyrics.KMP/shared/src/commonMain/kotlin/com/jayfunc/betterlyrics/models/settings/AppSettings.kt
package com.jayfunc.betterlyrics.models.settings

class AppSettings {
    var version: String = ""
    var translationSettings: TranslationSettings? = null
    var generalSettings: GeneralSettings? = null
    var musicGallerySettings: MusicGallerySettings? = null
    var advancedSettings: AdvancedSettings? = null
    var lyricsSaveConfig: Any? = null
    var systemTraySettings: SystemTraySettings? = null
    var localMediaFolders: List<Any> = emptyList()
    var mediaSourceProvidersInfo: List<Any> = emptyList()
    var mappedSongSearchQueries: List<Any> = emptyList()
    var windowBoundsRecords: List<Any> = emptyList()
    var starredPlaylists: List<Any> = emptyList()
    var pluginsInfo: List<Any> = emptyList()
    var lyricsCardConfigs: List<Any> = emptyList()
    var layoutProfiles: List<Any> = emptyList()
}
